### ISSUE TITLE
docusaurus doesn't run on bun

### ISSUE THREAD

**Issue by noraj** [facebook/docusaurus#9294]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

The current architecture is so that docusaurus requires nodejs to run while it could easily run on other javascript servers like bun.

### Reproducible demo

_No response_

### Steps to reproduce

- bun install
- docusaurus start

### Expected behavior

Working with bun as well with nodejs

### Actual behavior

```
➜ bun start                                                                                                                                                                                                          
$ docusaurus start                                                                                                                                                                                                   
/home/noraj/Dev/rawsec-cybersecurity-inventory/docusaurus/node_modules/@docusaurus/utils/lib/index.js:9                                                                                                              
exports.replaceMarkdownLinks = exports.writeMarkdownHeadingId = exports.parseMarkdownString = exports.parseMarkdownContentTitle = exports.parseFrontMatter = exports.createExcerpt = exports.parseMarkdownHeadingId =
 exports.groupTaggedItems = exports.normalizeFrontMatterTags = exports.buildSshUrl = exports.buildHttpsUrl = exports.hasSSHProtocol = exports.removeTrailingSlash = exports.addTrailingSlash = exports.addLeadingSlas
h = exports.resolvePathname = exports.isValidPathname = exports.encodePath = exports.fileToPath = exports.getEditUrl = exports.normalizeUrl = exports.findAsyncSequential = exports.mapAsyncSequential = exports.remo
vePrefix = exports.removeSuffix = exports.localizePath = exports.getPluginI18nPath = exports.updateTranslationFileMessages = exports.mergeTranslations = exports.GitNotFoundError = exports.FileNotTrackedError = exp
orts.getFileCommitDate = exports.readOutputHTMLFile = exports.generate = exports.WEBPACK_URL_LOADER_LIMIT = exports.DEFAULT_PLUGIN_ID = exports.DEFAULT_PORT = exports.CODE_TRANSLATIONS_FILE_NAME = exports.DEFAULT_
I18N_DIR_NAME = exports.THEME_PATH = exports.OUTPUT_STATIC_ASSETS_DIR_NAME = exports.DEFAULT_STATIC_DIR_NAME = exports.SRC_DIR_NAME = exports.GENERATED_FILES_DIR_NAME = exports.BABEL_CONFIG_FILE_NAME = exports.DEF
AULT_CONFIG_FILE_NAME = exports.DEFAULT_BUILD_DIR_NAME = exports.DOCUSAURUS_VERSION = exports.NODE_MINOR_VERSION = exports.NODE_MAJOR_VERSION = void 0;                                                              
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
TypeError: Cannot set property NODE_MAJOR_VERSION of #<Object> which has only a getter                                                                                                                               
    at Object.<anonymous> (/home/noraj/Dev/rawsec-cybersecurity-inventory/docusaurus/node_modules/@docusaurus/utils/lib/index.js:9:1421)                                                                             
    at Module._compile (node:internal/modules/cjs/loader:1241:14)                                                                                                                                                    
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)                                                                                                                                             
    at Module.load (node:internal/modules/cjs/loader:1091:32)                                                                                                                                                        
    at Module._load (node:internal/modules/cjs/loader:938:12)                                                                                                                                                        
    at Module.require (node:internal/modules/cjs/loader:1115:19)                                                                                                                                                     
    at require (node:internal/modules/helpers:130:18)                                                                                                                                                                
    at Object.<anonymous> (/home/noraj/Dev/rawsec-cybersecurity-inventory/docusaurus/node_modules/@docusaurus/utils/lib/dataFileUtils.js:15:17)                                                                      
    at Module._compile (node:internal/modules/cjs/loader:1241:14)                                                                                                                                                    
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)                                                                                                                                             
                                                                                                                                                                                                                     
Node.js v20.6.0                                                                                                                                                                                                      
error: script "start" exited with code 1 (SIGHUP)      
```

### Your environment

- Docusaurus version used: 2.4.1
- Environment name and version : bun 1.0.0
- Operating system and version : archlinux


### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Your error stack references Node, so not sure how it's related to bun. This error is a duplicate of https://github.com/facebook/docusaurus/issues/9291 and fixed by the latest Node.

**Comment by noraj** [comment]
@Josh-Cena Sorry, it was an error of environment. I modified my original message with a reproducible clean environment.

**Comment by Josh-Cena** [comment]
Responded in oven-sh/bun#4742.

**Comment by slorber** [comment]
There's a dedicated Bun issue there, let's use it: https://github.com/oven-sh/bun/issues/3426

**Comment by slorber** [comment]
For reference this is why we use the Bun issue instead of the Docusaurus issue:

<img width="686" alt="CleanShot 2023-09-12 at 18 32 58@2x" src="https://github.com/facebook/docusaurus/assets/749374/747db891-cc84-4698-a999-c18098bd689b">

https://twitter.com/jarredsumner/status/1700497869160268155

**Comment by noraj** [comment]
I agree with you and I'm sorry for opening this issue there.

What's deceptive and misleading is the release of Bun v1.0.0. Normally following semver 0.x.x are kind of unstable version while 1.0.0 means the 1st stable version. When I saw and tested v1 I expected Bun to be production ready while in fact it's still in beta since they claim to want full Node.js compatibility and in v1 the Node.js compatibility is still vastly incomplete https://bun.sh/docs/runtime/nodejs-apis.

So yeah most errors come from the fact that the Node.js compatibility is not 100% yet and not because upstream project have hard-coded nodejs stuff.

**Comment by slorber** [comment]
Not sure it's their fault, it's also because Docusaurus uses a package that depends on V8

![CleanShot 2023-09-13 at 18 09 59@2x](https://github.com/facebook/docusaurus/assets/749374/3e213841-c1ee-4e84-b90e-ba8b07d4f37a)
